hazelcast/proxy/map.py

```diff
@@ -186,7 +187,7 @@ def add_entry_listener(
             LOADED=loaded_func,
         )
 
-        if key and predicate:
+        if key is not None and predicate:
```

Comment:
Why do we have this asymmetry? key and predicate either should be allowed to be null or not.

Reply:
predicate can be null or be a valid predicate. it can't be for example 0 or False, so there is no need for is None checks for it

Reply:
I think predicates are not allowed to be null in Java. So you should change the default None value to a sentinel.